File: backend/app/schemas/task.py
from typing import Optional

from pydantic import BaseModel


class TaskOut(BaseModel):
    id: int
    project_id: str
    milestone_id: Optional[int] = None
    parent_task_id: Optional[int] = None
    title: str
    description: Optional[str] = None
    assignee: Optional[str] = None
    role_key: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    est_hours: float = 0.0
    actual_hours: float = 0.0
    status: Optional[str] = None
    progress_pct: float = 0.0
    priority: Optional[str] = None
    jira_key: Optional[str] = None
    sort_order: Optional[int] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

File: frontend/src/types/task.ts
export interface Task {
  id: number;
  project_id: string;
  milestone_id: number | null;
  parent_task_id: number | null;
  title: string;
  notes: string | null; // Rich HTML (renamed from description in v7)
  assignee: string | null;
  role_key: string | null;
  start_date: string | null;
  end_date: string | null;
  est_hours: number;
  actual_hours: number;
  status: string | null;
  progress_pct: number;
  priority: string | null;
  jira_key: string | null;
  sort_order: number | null;
  created_at: string | null;
  updated_at: string | null;
  updated_by: string | null;
}
